cmake/SociConfig.cmake

```diff
@@ -69,7 +69,7 @@ else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
         elseif (CMAKE_SYSTEM_NAME MATCHES "Windows")
             # We need to disable the warnings for using %I64d with MinGW.
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format -Wno-format")
```

Comment:
I found the reason for the mysterious discrepancy, i.e. why the same compiler with the same options compiled my test without problems but choked on SOCI code: this is due to `__USE_MINGW_ANSI_STDIO` being defined for SOCI but not for the simple example. And the reason for this is that including any C++ standard library headers ends up including `lib\gcc\i686-w64-mingw32\4.8.3\include\c++\i686-w64-mingw32\bits\os_defines.h` which contains
```
// Make sure that POSIX printf/scanf functions are activated.  As
// libstdc++ depends on POSIX-definitions of those functions, we define
// it unconditionally.
#undef __USE_MINGW_ANSI_STDIO
#define __USE_MINGW_ANSI_STDIO 1
```

So, long story short, the original fix replacing `%I64d` with `%lld` was in fact correct as the latter is what works with MinGW ANSI stdio implementation, while the former only works with the MSVC non-conforming-but-nevertheless-used-by-default-unless-a-standard-header-is-included (!) implementation.

Sorry for the confusion, I had no idea it was so twisted.